pkg/controllers/mcs/multiclusterservice_controller.go

```diff
@@ -0,0 +1,620 @@
+				*metav1.NewControllerRef(ownerSvc, ownerSvc.GroupVersionKind()),
+			},
+			Finalizers: []string{MultiClusterServiceControllerFinalizer},
+		},
+	}
+	oldSvcImport := &mcsv1alpha1.ServiceExport{}
+	err := c.Client.Get(context.TODO(), types.NamespacedName{
+		Name:      mcs.Name,
+		Namespace: mcs.Namespace,
+	}, oldSvcImport)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+		err = c.Client.Create(context.TODO(), newSvcExport.DeepCopy())
+		if err != nil {
+			msg := fmt.Sprintf("Create derived ServiceExport(%s/%s) failed, Error: %v", newSvcExport.Namespace, newSvcExport.Name, err)
+			klog.Errorf(msg)
+			c.EventRecorder.Eventf(mcs, corev1.EventTypeWarning, events.EventSyncDerivedServiceExportFailed, msg)
+			return err
+		}
+		msg := fmt.Sprintf("Sync the serviceExport of MultiClusterService(%s/%s) successful", mcs.Namespace, mcs.Name)
+		klog.V(4).Infof(msg)
+		c.EventRecorder.Eventf(mcs, corev1.EventTypeNormal, events.EventSyncDerivedServiceExportSucceed, msg)
+		return c.ensureServiceExportWork(mcs, newSvcExport)
+	}
+
+	if !metav1.IsControlledBy(oldSvcImport, ownerSvc) {
+		err = c.Client.Update(context.TODO(), newSvcExport)
+		if err != nil {
+			klog.Errorf("Update derived ServiceExport(%s/%s) failed, Error: %v", newSvcExport.Namespace, newSvcExport.Name, err)
+			return err
+		}
+	}
+	return c.ensureServiceExportWork(mcs, newSvcExport)
+}
+
+func (c *MultiClusterServiceController) ensureServiceExportWork(mcs *networkingv1alpha1.MultiClusterService, svcExport *mcsv1alpha1.ServiceExport) error {
+	svcRB := &workv1alpha2.ResourceBinding{}
+	if err := c.Client.Get(context.TODO(), types.NamespacedName{Namespace: mcs.Namespace, Name: names.GenerateBindingName(util.ServiceKind, mcs.Name)}, svcRB); err != nil {
+		if apierrors.IsNotFound(err) {
+			msg := fmt.Sprintf("the service(%s/%s) has never been propagated to member clusters", mcs.Namespace, mcs.Name)
+			klog.Warningf(msg)
+			c.EventRecorder.Eventf(svcExport, corev1.EventTypeWarning, events.EventReasonSyncDerivedWorkFailed, msg)
+			return nil
+		}
+		return err
+	}
+	for _, targetCluster := range svcRB.Spec.Clusters {
+		key, err := keys.FederatedKeyFunc(targetCluster.Name, svcExport)
+		if err != nil {
+			klog.Warningf("Failed to generate work key for ServiceExport(%s/%s) to cluster(%s)", mcs.Namespace, mcs.Name, targetCluster.Name)
+			continue
+		}
+		c.WorkProcessor.Add(key)
+	}
+	return nil
+}
+
+func (c *MultiClusterServiceController) reconcileServiceExport(mcs *networkingv1alpha1.MultiClusterService) error {
+	if !mcs.Spec.HasExposureTypeCrossCluster() || !mcs.DeletionTimestamp.IsZero() {
+		return c.deleteDerivedServiceExport(mcs)
+	}
+	ownerSvc := &corev1.Service{}
+	err := c.Client.Get(context.TODO(), types.NamespacedName{
+		Name:      mcs.Name,
+		Namespace: mcs.Namespace,
+	}, ownerSvc)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return c.deleteDerivedServiceExport(mcs)
+		}
+		return err
+	}
```

Comment:
We have added OwnerReferences to the ServiceExport object. Can we remove the processing here?

Reply:
When only deleting `mcs` without deleting `svc`, it is also necessary to delete `ServiceExport`, so it is not simply relying on `OwnerReferences` to delete `ServiceExport`.